Makefile

```diff
@@ -0,0 +1,153 @@
+UNCONVERT_CHECK := $(shell command -v unconvert 2> /dev/null)
+INEFFASSIGN_CHECK := $(shell command -v ineffassign 2> /dev/null)
+MISSPELL_CHECK := $(shell command -v misspell 2> /dev/null)
+ERRCHECK_CHECK := $(shell command -v errcheck 2> /dev/null)
+UNPARAM_CHECK := $(shell command -v unparam 2> /dev/null)
+GOCYCLO_CHECK := $(shell command -v gocyclo 2> /dev/null)
+
+all: get-tools get-vendor-deps install
+
+########################################
+### CI
+
+ci: get-tools get-vendor-deps install
+
+########################################
+### Build
+
+# This can be unified later, here for easy demos
+build:
+ifeq ($(OS),Windows_NT)
+	go build $(BUILD_FLAGS) -o build/ethermint.exe ./*.go
+else
+	go build $(BUILD_FLAGS) -o build/ethermint ./*.go
+endif
+
+install:
+	go install $(BUILD_FLAGS) ./*.go
+
+########################################
+### Tools & dependencies
+
+update-tools:
+	@echo "Updating golang dependencies"
+	go get -u -v $(DEP) $(GOLINT) $(GOMETALINTER) $(UNCONVERT) $(INEFFASSIGN) $(MISSPELL) $(ERRCHECK) $(UNPARAM) $(GOCYCLO)
+
+get-tools:
+	ifdef DEP_CHECK
+		@echo "Dep is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing dep"
+		go get -v $(DEP)
+	endif
+	ifdef GOLINT_CHECK
+		@echo "Golint is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing golint"
+		go get -v $(GOLINT)
+	endif
+	ifdef GOMETALINTER_CHECK
+		@echo "Gometalinter.v2 is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing gometalinter.v2"
+		go get -v $(GOMETALINTER)
+	endif
+	ifdef UNCONVERT_CHECK
+		@echo "Unconvert is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing unconvert"
+		go get -v $(UNCONVERT)
+	endif
+	ifdef INEFFASSIGN_CHECK
+		@echo "Ineffassign is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing ineffassign"
+		go get -v $(INEFFASSIGN)
+	endif
+	ifdef MISSPELL_CHECK
+		@echo "misspell is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing misspell"
+		go get -v $(MISSPELL)
+	endif
+	ifdef ERRCHECK_CHECK
+		@echo "errcheck is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing errcheck"
+		go get -v $(ERRCHECK)
+	endif
+	ifdef UNPARAM_CHECK
+		@echo "unparam is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing unparam"
+		go get -v $(UNPARAM)
+	endif
+	ifdef GOYCLO_CHECK
+		@echo "goyclo is already installed.  Run 'make update-tools' to update."
+	else
+		@echo "Installing goyclo"
+		go get -v $(GOCYCLO)
+	endif
+
+get-vendor-deps:
+	@rm -rf vendor/
+	@echo "--> Running dep ensure"
+	@dep ensure -v
+
+
+########################################
+### Documentation
+
+godocs:
+	@echo "--> Wait a few seconds and visit http://localhost:6060/pkg/github.com/cosmos/ethermint"
+	godoc -http=:6060
+
+########################################
```

Comment:
Let's remove the testing stuff for now until we need it 👍

Reply:
Good call.